actionpack/lib/action_dispatch/middleware/server_timing.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require "action_dispatch/middleware/server_timing_subscriber"
+
+module ActionDispatch
+  class ServerTiming
+    class << self
+      def add_timing(key:, dur: nil, desc: nil)
+        timings[key] = { dur: dur, desc: desc }
+      end
+
+      def to_header
+        headers = []
```

Comment:
I think using the spec's language this should be called `metrics`. Several metrics can be conveyed on a single header: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Server-Timing#Syntax

Reply:
nice! I was struggling to pick a good name.